base/reduce.jl

```diff
@@ -306,25 +306,29 @@ with reduction `op` over an empty array with element type of `T`.
 
 If not defined, this will throw an `ArgumentError`.
 """
-reduce_empty(op, T) = _empty_reduce_error()
-reduce_empty(::typeof(+), T) = zero(T)
+reduce_empty(op, ::Type{T}) where T = _empty_reduce_error()
```

Comment:
Style nit: I think we usually write short form functions with the braces `{}` to visually distinguish them:
```suggestion
reduce_empty(op, ::Type{T}) where {T} = _empty_reduce_error()
```